thrift/cadence.thrift

```diff
@@ -108,6 +108,7 @@ service WorkflowService {
       6: shared.LimitExceededError limitExceededError,
       7: shared.EntityNotExistsError entityNotExistError,
       8: shared.ClientVersionNotSupportedError clientVersionNotSupportedError,
+      9: shared.WorkflowExecutionAlreadyCompletedError workflowExecutionAlreadyCompletedError,
```

Comment:
Hmm, I think this should match the errors thrown by the history service. Same for SignalWithStartWorkflowExecution.

I understand that currently StartWorkflowExecution and SignalWithStartWorkflowExecution may return ErrAlreadyCompleted when workflowID reuse policy is TerminateIfRunning, but this looks like a bug to me, and those two APIs should not return entityNotExist or workflowAlreadyCompleted.

Reply:
Call stack for already completed error is as following: 
```
- Service/history/historyEngine.go StartWorkflowExecution
	- Service/history/historyEngine.go startWorkflowHelper
		- Service/history/historyEngine.go terminateAndStartWorkflow
			 - UpdateWorkflowLoop
			      if !runningMutableState.IsWorkflowExecutionRunning() {
			              return nil, workflow.ErrAlreadyCompleted
		              }
```

Reply:
Seems like this would only happen in a race condition where a workflow is in running state when first checked but is no longer in that state by the time we are trying to terminate it. I think the right thing to do here is to handle the error and proceed as if it was completed when first checked.